ADR/0055-revised-component-model.md

```diff
@@ -0,0 +1,628 @@
+
+## References
+
+* [Konflux Community Call, 2025-03-11](https://www.youtube.com/watch?v=IwNFRNjRC_Q)
+* [Konflux Community Call, 2025-03-25](https://www.youtube.com/watch?v=VwzDbUH_vjU)
+
+## Context
+
+The existing application and component model in Konflux presents several
+limitations that hinder our ability to scale and support flexible workflows.
+Specifically, the current model requires a new component to be registered for
+each use case, even when the same underlying code is being used in multiple
+release scenarios. This creates unnecessary overhead and complexity, while also
+limiting the reusability of our components.
+
+Furthermore, ownership of the current APIs is unclear. They were originally
+owned by HAS (which is now deprecated) and exhibit excessive coupling across
+multiple services.
+
+## Decision
+
+We have decided to adopt a revised component model that replaces the concept of
+an "application" with a "component group." The key elements of this new model
+are:
+
+* **Component Branch**: A component can have one or more component branches,
+  which are a combination of a component and a specific branch or tag. A single
+  component branch can be referenced by multiple component groups.
+  build-service owns the Component API going forwards and the spec changes
+  for branches.
+* **Component Group**: A component group can contain other component groups or
+  component branches, forming a Directed Acyclic Graph (DAG) that represents
+  the release structure. This allows for greater flexibility and reusability,
+  as a component can be used in multiple groups without re-registration.
+  integration-service owns the new Group API and the eventual decommissioning
+  of the old Application API.
+
+The implementation will be phased, starting with the core build team and
+gradually migrating services and users over time, with the final goal of
+decommissioning the old model once the migration is complete.
+
+### Detail of new Component CR spec
+
+```
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  name: example
+  namespace: example
+
+spec: # Required field
+  # specific actions will be processed and then removed
+  actions:
+    # specify names of component versions to restart the push build for
+    # both can be specified at same time, but eliminate duplicates
+    trigger-push-build: Test
+    trigger-push-builds:
+      - Test
+      - Different pipeline
+
+    # Only for build pipeline configuration PR explicit creation,
+    # otherwise if version exists in the .spec.source.versions
+    # and status doesn't have it, do onboarding without PR.
+    # When create-pipeline-configuration-pr is specified,
+    # at least one of 'all_versions', 'version', 'versions' has to be specified.
+    # 'all_versions' has precedence.
+    # When both 'version' and 'versions' will be specified use all versions from them
+    # but eliminate duplicates.
```

Comment:
I think we need to:
 - Convert the phrases to sentences.
 - Follow common comment style.
 - Fix grammar / punctuation.
The same applies to further comments.